docs/Middleware/Ajax.md

```diff
@@ -0,0 +1,22 @@
+# Ajax Middleware
+This middleware provides an alternative to the component, and is intended for applications which have migrated to the
+middleware stack provided in CakePHP 3.5 and above.
+Features and configuration are intended to be the same between the two methods, you should only need one or
+the other.
+Refer to [Component documentation](../Component/Ajax.md) for those details.
+
+## Usage
+Load the Ajax middleware:
+```php
+$middlewareQueue->->add(AjaxMiddleware::class);
+```
+
+As with the component, you can pass the settings either directly inline here, or use Configure to set them globally.
+
+```php
+$middlewareQueue->->add(new AjaxMiddleware(['viewClass' => 'App.Ajax']))
```

Comment:
Do we need App. prefix? I thought just Ajax works.

Reply:
Yep. Changed to "MyAjax" to make it clear that it's a user-provided class.